cpp/DataStorm/customEncoding/TimePoint.h

```diff
@@ -0,0 +1,50 @@
+#ifndef TIME_POINT_H
+#define TIME_POINT_H
+
+#include <DataStorm/DataStorm.h>
+#include <Ice/Ice.h>
+
+namespace DataStorm
```

Comment:
Moved the Encoded/Decoder template from Reader/Writer to this file.

I also documented them and updated to use bitwise operators instead of division which I find much clear for a binary encoding.